src/Rx/Weno/file_download.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+/**
+ * Weno Zipped file download
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Kofi Appiah <kkappiah@medsov.com>
+ * @copyright Copyright (c) 2023 omega systems group international <info@omegasystemsgroup.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+require_once("../../../interface/globals.php");
+
+use OpenEMR\Common\Crypto\CryptoGen;
+use OpenEMR\Rx\Weno\LogDataInsert;
+use OpenEMR\Common\Logging\EventAuditLogger;
+
+$cryptogen = new CryptoGen();
+$weno_username = $GLOBALS['weno_admin_username'];
+$weno_password = $GLOBALS['weno_encryption_key'];
+$encryption_key = $cryptoGen->decryptStandard($GLOBALS['weno_encryption_key']);
+$baseurl = "https://online.wenoexchange.com/en/EPCS/DownloadPharmacyDirectory";
+
+$data = array(
+    "UserEmail"             => $weno_username,
+    "MD5Password"           => md5($weno_password),
+    "ExcludeNonWenoTest"    => "N",
+    "Daily"                 => "Y"
+);
+
+if (date("l") == "Sunday") { //if today is Sunday download the weekly file
+    $data["Daily"] = "N";
+}
+
+$json_object = json_encode($data);
+$method = 'aes-256-cbc';
+
+$key = substr(hash('sha256', $encryption_key, true), 0, 32);
+
+$iv = chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0);
+
+$encrypted = base64_encode(openssl_encrypt($json_object, $method, $key, OPENSSL_RAW_DATA, $iv));
```

Comment:
wouldn't consider this secure from a encryption perspective, although not sure if that is necessary since this looks like it is processed on weno's side. If need it to be secure from a encryption perspective, then would need 2 things:
1. a random IV
2. incorporation of a hash_hmac (can see the coreEncrypt() function for example of how to do this)

Reply:
@bradymiller, according to the Weno team, IV is an Initialization Vector and everything is handled at Weno's end.

A registered IP is required at Weno's end before processing the request.  The encryption key is also generated at Weno's end for the Admin user.